### PR TITLE
Update for beta 16

### PR DESCRIPTION
**Pull request by askvortsov1**
- Copy dot files directly
- Include testing infrastructure
- Include prettier stuff
- Add automated actions
- Update README
- Update package.json
- Remove flagrow from extension generator

### CODE REVIEW

**Comment by askvortsov1** [comment]
This will be followed up by https://github.com/askvortsov1/extension-generator/pull/1

File: boilerplate/js/package.json [hunk 1]
```diff
@@ -7,8 +7,13 @@
...
     "webpack-cli": "^3.0.7"
   },
+  "devDependencies": {
+    "prettier": "2.0.2"
+  },
   "scripts": {
     "dev": "webpack --mode development --watch",
-    "build": "webpack --mode production"
+    "build": "webpack --mode production",
+    "format": "prettier --write src",
```
Comment (davwheat) [inline boilerplate/js/package.json:16]: Why don't we allow Prettier to format the entire extension, including Less, YML, JSON, etc? It'd help to maintain styles even more, especially as there is currently nothing in place for these.

Comment (askvortsov1) [reply boilerplate/js/package.json:16]: I don't think the JS portion of the extension should be responsible for formatting anything outside of the JS portion of the extension. We could do this, but then we should move the formatting task to `composer.json`.

Comment (davwheat) [reply boilerplate/js/package.json:16]: That would probably be a good idea.

Might it be worth using Prettier to format the PHP, too? There's an [officially maintained plugin for PHP formatting](https://github.com/prettier/plugin-php), and that means we wouldn't be combined different tools (like StyleCI) with Prettier.

We'd need to make sure the user installs `prettier` and `@prettier/plugin-php` globally, of course. (Without the plugin, Prettier will complain that no formatters exist for `.php` files)

Comment (askvortsov1) [reply boilerplate/js/package.json:16]: If we want to do that, I think we should consider it as a separate issue from the stuff here: my goal with this PR is to bring the template up to date with recent core changes.

**Pull request closed without merging**